pinot-core/src/main/java/org/apache/pinot/core/plan/TransformPlanNode.java

```diff
@@ -89,39 +73,36 @@ private void extractColumnsAndTransforms(@Nonnull BrokerRequest brokerRequest,
         for (String expression : brokerRequest.getGroupBy().getExpressions()) {
           TransformExpressionTree transformExpressionTree = TransformExpressionTree.compileToExpressionTree(expression);
           transformExpressionTree.getColumns(_projectionColumns);
-          _expressionTrees.add(transformExpressionTree);
+          _expressions.add(transformExpressionTree);
         }
       }
     } else {
       Selection selection = brokerRequest.getSelections();
       // No ordering required, select minimum number of documents
       if (!selection.isSetSelectionSortSequence()) {
-        _maxDocPerNextCall = Math.min(selection.getOffset() + selection.getSize(), _maxDocPerNextCall);
+        _maxDocPerNextCall = Math.max(Math.min(selection.getSize(), _maxDocPerNextCall), 1);
```

Comment:
why is this Math being changed? how is the behaviour staying the same with this?

Reply:
For selection only queries, offset does not matter because no order is provided. We need to fetch at least one document to prevent exception (EmptySelectionOperator use TransformOperator to determine the expression type). These behavior are checked in the test

Reply:
Added comments for that